### ISSUE TITLE
FN neg_multiply:  x *= -1 ?

### ISSUE THREAD

**Issue by matthiaskrgr** [rust-lang-nursery/rust-clippy#3037]
For some reason the lint catches
```rust
x = x * -1;
````
but not
````rust
x *= -1;
````
is this intended?
(I assume it wants ````x = -x````)

**Comment by felix91gr** [comment]
This false negative still persists: https://play.rust-lang.org/?version=stable&mode=debug&edition=2024&gist=97d60e437c5fea498155ef14e93b4d61

Though as @fhartwig said in #3375, it's not clear whether it makes sense to change this or not, since it's a 1 character difference in length.

Perhaps it still makes sense, since multiplying by -1 masks the intent a little bit (which is purely to **negate a value**)?

Phil Hansch has already added a S-needs-discussion label, I assume exactly because of this.

Anyway. Wanted to test this issue and write a report about it. Hope this helps one day.
